src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -381,12 +381,8 @@ void TimeSynchronizationServer::OnTimeSyncCompletionFn(TimeSourceEnum timeSource
         }
         return;
     }
-    mGranularity  = granularity;
-    Status status = TimeSource::Set(kRootEndpointId, timeSource);
-    if (!(status == Status::Success || status == Status::UnsupportedAttribute))
-    {
-        ChipLogError(Zcl, "Writing TimeSource failed.");
-    }
+    mGranularity = granularity;
+    mTimeSource  = timeSource;
```

Comment:
Changes to the value of the attribute need to mark it dirty, no?  This applies to a bunch of the attributes in this cluster, actually: it seems to be completely broken wrt subscriptions....